sklearn/preprocessing/_encoders.py

```diff
@@ -558,6 +558,10 @@ class OneHotEncoder(_BaseEncoder):
           `handle_unknown='ignore'`. Infrequent categories exist based on
           `min_frequency` and `max_categories`. Read more in the
           :ref:`User Guide <encoder_infrequent_categories>`.
+        - 'warn' : When an unknown category is encountered during
+          transform a warnining will be issued and the resulting one-hot encoded
+          columns for this feature will be all zeros. In the inverse transform,
+          an unknown category will be denoted as None.
```

Comment:
The above suggestion would need to be updated to replace `ignore` by `infrequent_if_exist` if you take my main feedback into account.

Reply:
Then maybe a cleaner solution would be adding 'verbose' parameter? This way user will get warnings and other useful information no matter what option 'handle_unknown' is chosen.

Reply:
No strong opinion. But if we use a verbose parameter, it should be a simple call to `print` (on stdout) rather than a warning.

Reply:
@glemaitre How should I proceed?

Reply:
Personally, I would prefer the warning because of the type raise and it avoids introducing an additional parameter. I would rather wait to see if in the future, we really end-up having to show additional information to the user where it would then make sense to have a verbose option.